lib/tapioca/gem/pipeline.rb

```diff
@@ -152,11 +158,15 @@ def on_constant(event)
         return if alias_namespaced?(name)
         return if seen?(name)
 
-        constant = event.constant
-        return if T::Enum === constant # T::Enum instances are defined via `compile_enums`
+        return if T::Enum === event.constant # T::Enum instances are defined via `compile_enums`
 
         mark_seen(name)
-        compile_constant(name, constant)
+
+        if event.is_a?(Gem::NotOwnedConstantFound)
+          compile_not_owned_constant(name, event.constant)
+        else
+          compile_constant(name, event.constant)
+        end
```

Comment:
Can we turn this into a `case/when` to make it explicit that we are matching on the type of `event`?
```suggestion
        case event
        when Gem::NotOwnedConstantFound
          compile_not_owned_constant(name, event.constant)
        else
          compile_constant(name, event.constant)
        end
```